DigitalLearningSolutions.Web/Controllers/TrackingSystem/Delegates/BulkUploadController.cs

```diff
@@ -34,5 +44,58 @@ public IActionResult DownloadDelegates()
                 $"DLS Delegates for Bulk Update {DateTime.Today:yyyy-MM-dd}.xlsx"
             );
         }
+
+        [Route("StartUpload")]
+        [HttpGet]
+        public IActionResult StartUpload()
+        {
+            TempData.Clear();
+            var model = new UploadDelegatesViewModel();
+            return View("StartUpload", model);
+        }
+
+        [Route("StartUpload")]
+        [HttpPost]
+        public IActionResult StartUpload(UploadDelegatesViewModel model)
+        {
+            model.ClearDateIfNotSendEmail();
+
+            if (!ModelState.IsValid)
+            {
+                return View("StartUpload", model);
+            }
+
+            try
+            {
+                var results = delegateUploadFileService.ProcessDelegatesFile(
+                    model.DelegatesFile!,
+                    User.GetCentreId(),
+                    model.GetWelcomeEmailDate()
+                );
+                TempData.Set(results);
+                return RedirectToAction("UploadCompleted");
+            }
+            catch (InvalidHeadersException)
+            {
+                return RedirectToAction("UploadFailed");

```

Comment:
Any reason we're redirecting over returning a view (is it because of the back button functionality)?